coalib/misc/Caching.py

```diff
@@ -0,0 +1,142 @@
+import os
+import calendar
+import time
+import appdirs
+import pickle
```

Comment:
@fneu I think it will, at worst, corrupt the file. I didn't think about the possibility of that happening.

So I think, as @Makman2 said, adding a UnpickingError should suffice. The worst that will happen in that case is that every project will need to be cached again (maybe we can display an error saying caching broke, so every project needs to be cached again). I don't think that's *too* bad.